src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -303,6 +305,8 @@ protected internal virtual void WriteMetricEvents(SystemMetricEvent[] metricEven
 
         protected virtual void Dispose(bool disposing)
         {
+            _logger.LogDebug($"Disposing {nameof(MetricsEventManager)}");
```

Comment:
If the factory that creates this logger is disposed prior to this type, we might run into issues here, right? Not sure how likely this is to happen, but it might be good to be defensive.

Reply:
The LoggerFactory (and it's LoggerProviders) will be disposed after this as this takes a dependency on it via the constructor. But I can wrap it in try/catch to be extra careful.